### DISCUSSION TITLE
How to clear the contents of `Form` when closing `Drawer` or `Modal`?

### DISCUSSION THREAD

**Discussion by ZCY-G** [mantinedev/mantine#4155] [Q&A]
I am using `Form` in `Drawer` and `Modal` without `keepMounted` set, can you provide an API to automatically uninstall the `Form` component when I close `Drawer` or `Modal`, like [Ant Design](https://ant.design/components/modal#api) as follows:

There is a situation that using `<Modal />` with Form, which won't clear fields value when closing Modal even you have set `destroyOnClose`. You need `<Form preserve={false} />` in this case.

**Comment by rtivital** [comment]
What is `Form` component? Mantine does not provide it.

**Reply by ZCY-G** [reply]
Sorry, I was wrong, it's using the native `form` in conjunction with the `use-form` hook.

**Reply by rtivital** [reply]
Then you can call `form.reset` when Drawer is closed, you can do that in useEffect or your close handler.

**Reply by ZCY-G** [reply]
That's what I'm doing now, but I wanted something simpler to write, rather than writing an extra method just to reset the form.
